src/PKSim.BatchTool/Program.cs

```diff
@@ -1,34 +1,44 @@
 ﻿using System;
 using System.Windows.Forms;
+using OSPSuite.Presentation.Services;
 using OSPSuite.Utility.Container;
 using OSPSuite.Utility.Extensions;
 using PKSim.BatchTool.Presenters;
-using OSPSuite.Presentation.Services;
 
 namespace PKSim.BatchTool
 {
+   [Flags]
+   enum ExitCodes
+   {
+      Success = 0,
+      Error = 1 << 0,
+   }
+
    static class Program
    {
       /// <summary>
-      /// The main entry point for the application.
+      ///    The main entry point for the application.
       /// </summary>
       [STAThread]
-      static void Main(string[] args)
-      { 
+      static int  Main(string[] args)
+      {
          Application.EnableVisualStyles();
          Application.SetCompatibleTextRenderingDefault(false);
          try
-         { 
+         {
             BatchStarter.Start();
             var mainPresenter = IoC.Resolve<IBatchMainPresenter>();
-            mainPresenter.Initialize(BatchStartOptions.From(args));
-            Application.Run(mainPresenter.View.DowncastTo<Form>());
+            var options = BatchStartOptions.From(args);
+            mainPresenter.Initialize(options);
+            Application.Run(mainPresenter.ActiveView.DowncastTo<Form>());
```

Comment:
If you called mainPresenter.StartBatchRun instead of Application.Run then wouldn't you have a 'headless' mode? Again, not sure how the batch runner works.

All depends if you can get away without a message loop.